src/cli/multiopenjscad.js

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env node
+
+#a simple script that reads an interactive params example and generates every variation
+#to use this:  node multiopenjscad.js /path/to/gears.jscad
+#you must have the packages mkdirp and eval installed to use this.
+const fs = require('fs')
+const path = require('path')
+const {execSync} = require('child_process')
+var _eval = require('eval')
+var mkdirp = require('mkdirp');
+
+
+const args = process.argv.splice(2)
+if(args.length < 1){
+  console.log("pass file to parse")
+  process.exit()
+}
+
+inputFile = args[0]
+ftype = inputFile.split("/").slice(-1)[0].split(".")[0]
+console.log("class: "+ftype)
+let src = fs.readFileSync(inputFile, 'UTF8')
+src += "\ngetParameterDefinitions()\n"
+eval(src)
+function to_array(json){
+  array = []
+  if(json.hasOwnProperty('step')){
+    step = json['step']
+  }else{
+    step = 1
+  }
+  for(var i = json['min'];i <= json['max'];i+= step){
```

Comment:
the script requires min and max values to be specified for all params right ?
in other words : the params must be finite ?

Reply:
also favor let over var